### PR TITLE
Dependency updates 20200716

### LINKED ISSUE #6672: MockitoAnnotations.initMocks(Object) deprecated, blocks dependency update
###### Reproduction Steps

1. bump mockito-core dependency to 3.4.0
2. run `./gradlew jacocoUnitTestReport`

###### Expected Result

Clean run

###### Actual Result

Build fail as deprecation warnings are treated as errors (not a bad thing - it forces us to consider them carefully, like this) and `MockitoAnnotations.initMocks(Object)` has been deprecated but is used in a couple places

```
> Task :AnkiDroid:compileDebugUnitTestJavaWithJavac

C:\Users\travis\build\mikehardy\Anki-Android\AnkiDroid\src\test\java\com\ichi2\anki\AnalyticsTest.java:68: warning: [deprecation] initMocks(Object) in MockitoAnnotations has been deprecated

        MockitoAnnotations.initMocks(this);

                          ^

error: warnings found and -Werror specified

> Task :AnkiDroid:compileDebugUnitTestJavaWithJavac FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.

Use '--warning-mode all' to show the individual deprecation warnings.

See https://docs.gradle.org/6.5.1/userguide/command_line_interface.html#sec:command_line_warnings

25 actionable tasks: 1 executed, 24 from cache

C:\Users\travis\build\mikehardy\Anki-Android\AnkiDroid\src\test\java\com\wildplot\android\rendering\PieChartTest.java:44: warning: [deprecation] initMocks(Object) in MockitoAnnotations has been deprecated

        MockitoAnnotations.initMocks(this);

                          ^

C:\Users\travis\build\mikehardy\Anki-Android\AnkiDroid\src\test\java\com\wildplot\android\rendering\PieChartParameterizedTest.java:65: warning: [deprecation] initMocks(Object) in MockitoAnnotations has been deprecated

        MockitoAnnotations.initMocks(this);

                          ^

Note: C:\Users\travis\build\mikehardy\Anki-Android\AnkiDroid\src\test\java\com\ichi2\async\AbstractCollectionTaskTest.java uses unchecked or unsafe operations.

Note: Recompile with -Xlint:unchecked for details.

1 error

3 warnings
```

Related https://github.com/mikehardy/Anki-Android/pull/217

### PR DESCRIPTION
**Pull request by mikehardy**
## Purpose / Description
Keep track of upstream dependencies, slowly pulling in external change as we can vs as we have to

This is kind of a big one:

1. The jetbrains-annotations one was blocked by an external transitive dependency, which I use gradle to force (tested, working)
1. Mockito 3.4.0 has static mocking capabilities (yay!) so I converted away from PowerMock for everything
1. A gradle plugin update always sort of implies we all should upgrade Android Studio, so yeah get on that

## Fixes
Fixes #6672 

## Approach

1. I used a gradle resolutionStrategy to force our jetbrains-annotations version after an upstream PR looks like it's going nowhere
1. I used `Mockito.mockStatic()`to re-implement our unit tests that mocked static things

## How Has This Been Tested?

It's literally all tests and lint, so I was testing and linting all the time to do it :-)

## Learning (optional, can help others)
1. Upstream jetbrains-annotations PR: https://github.com/JakeWharton/timber/pull/398 / resolution strategy https://docs.gradle.org/current/dsl/org.gradle.api.artifacts.ResolutionStrategy.html
1. Mockito.mockStatic PR: https://github.com/mockito/mockito/pull/1955

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Comment by mikehardy** [comment]
re-pushed with adverse results of codacy all handled

File: AnkiDroid/src/test/java/com/ichi2/anki/ProductionCrashReportingTreeTest.java [hunk 1]
```diff
@@ -55,20 +54,30 @@ public void setUp() {
...
+            // set up the platform log so that if anyone calls these 2 methods at all, it throws
+            when(Log.v(anyString(), anyString(), any()))
+                    .thenThrow(new RuntimeException("Verbose logging should have been ignored"));
+            when(Log.d(anyString(), anyString(), any()))
+                    .thenThrow(new RuntimeException("Debug logging should be ignored"));
+            when(Log.i(anyString(), anyString(), any()))
+                    .thenThrow(new RuntimeException("Info logging should throw!"));
+
+            // now call our wrapper - if it hits the platform logger it will throw
+            try {
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/anki/ProductionCrashReportingTreeTest.java:67]: Optional: `AnkiAssert.doesNotThrow`

File: AnkiDroid/src/test/java/com/ichi2/anki/ProductionCrashReportingTreeTest.java [hunk 2]
```diff
@@ -55,20 +54,30 @@ public void setUp() {
...
+                Timber.v("verbose");
+                Timber.d("debug");
+            } catch (Exception e) {
+                Assert.fail("we were unable to log without exception?");
+            }
+
+            try {
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/anki/ProductionCrashReportingTreeTest.java:74]: Optional: Consider moving to `AnkiAssert`

File: AnkiDroid/src/test/java/com/ichi2/anki/ProductionCrashReportingTreeTest.java (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/test/java/com/ichi2/anki/ProductionCrashReportingTreeTest.java]: Yes! when I saw the codacy report saying I need to assert I vaguely remembered this facility but not enough to implement. I'd like it to become regular practice so I altered to use it in both cases

**Pull request merged**